src/internal/scheduler/QueueAction.ts

```diff
@@ -21,20 +20,21 @@ export class QueueAction<T> extends AsyncAction<T> {
   }
 
   public execute(state: T, delay: number): any {
-    return (delay > 0 || this.closed) ?
-      super.execute(state, delay) :
-      this._execute(state, delay) ;
+    return delay > 0 || this.closed ? super.execute(state, delay) : this._execute(state, delay);
   }
 
-  protected requestAsyncId(scheduler: QueueScheduler, id?: any, delay: number = 0): any {
+  protected requestAsyncId(scheduler: QueueScheduler, id?: TimerHandle, delay: number = 0): TimerHandle {
     // If delay exists and is greater than 0, or if the delay is null (the
     // action wasn't rescheduled) but was originally scheduled as an async
     // action, then recycle as an async action.
 
     if ((delay != null && delay > 0) || (delay == null && this.delay > 0)) {
       return super.requestAsyncId(scheduler, id, delay);
     }
+
     // Otherwise flush the scheduler starting with this action.
-    return scheduler.flush(this);
+    scheduler.flush(this);
+
+    return 1;
```

Comment:
Previously, this was returning `true`, which isn't a `TimerHandle` type. I changed it to a `1`. Perhaps less memory efficient, but it's code-efficient and in-character for a `TimerHandle`.

Reply:
Was this really returning `true`? AFAICT, `QueueScheduler` has no `flush`:

https://github.com/ReactiveX/rxjs/blob/4afbc16f053ffbfbd4f45d677056ff00d7711de7/src/internal/scheduler/QueueScheduler.ts#L3-L4

And the `flush` in `AsyncScheduler` - which `QueueScheduler` extends - returns `void`:

https://github.com/ReactiveX/rxjs/blob/4afbc16f053ffbfbd4f45d677056ff00d7711de7/src/internal/scheduler/AsyncScheduler.ts#L27

IDK what the consequences of returning something truthy will be when it previously appeared to return something falsy.